### PR TITLE
profiles/holoportos: implement Nginx dispatch 

### PR DESCRIPTION
**Pull request by zo-el**

### CODE REVIEW

File: profiles/holoportos/default.nix [hunk 1]
```diff
@@ -61,12 +61,69 @@ in
...
-      locations."/".root = pkgs.singletonDir "${./index.html}";
+      listen = [{
+        addr = "localhost";
+        port = 443;
+        ssl = true;
+        }];
+      locations = {
+        "/admin/" = {
+          root = pkgs.hp-admin-ui;
+          index = "index.html index.php";
```
Comment (yegortimoshenko) [inline profiles/holoportos/default.nix]: This is the default.
```suggestion
```

File: profiles/holoportos/default.nix [hunk 2]
```diff
@@ -61,12 +61,69 @@ in
...
+
     virtualHosts.default = {
       enableACME = true;
       onlySSL = true;
-      locations."/".root = pkgs.singletonDir "${./index.html}";
+      listen = [{
+        addr = "localhost";
+        port = 443;
+        ssl = true;
+        }];
```
Comment (yegortimoshenko) [inline profiles/holoportos/default.nix]: This entire section is a default and should be dropped.

File: profiles/holoportos/default.nix [hunk 3]
```diff
@@ -61,12 +61,69 @@ in
...
+
+        "/api/v1/"= {
+          proxyPass = "http://unix:/run/hpos-admin.sock:/";
+          extraConfig = ''
+            auth_request /auth;
+          '';
+        };
+
+        "/api/v1/ws/"= {
+          proxyPass = "http://127.0.0.1:42233";
```
Comment (yegortimoshenko) [inline profiles/holoportos/default.nix]:
```suggestion
          proxyPass = "http://localhost:42233";
```

File: profiles/holoportos/default.nix [hunk 4]
```diff
@@ -61,12 +61,69 @@ in
...
+          proxyWebsockets = true;
+        };
+
+        "/auth/"= {
+          proxyPass = "http://127.0.0.1:2884";
```
Comment (yegortimoshenko) [inline profiles/holoportos/default.nix]:
```suggestion
          proxyPass = "http://localhost:2884";
```

Comment (yegortimoshenko) [reply profiles/holoportos/default.nix]: Would be best to host this as Unix socket from the get-go.

File: profiles/holoportos/default.nix [hunk 5]
```diff
@@ -61,12 +61,69 @@ in
...
+          extraConfig = ''
+            internal;
+            proxy_set_header X-Original-URI $request_uri;
+          '';
+        };
+
+        # This is a place Holder
+        "/communities/"= {};
```
Comment (yegortimoshenko) [inline profiles/holoportos/default.nix]:
```suggestion
```

Let's drop this until it's implemented.

File: profiles/holoportos/default.nix [hunk 6]
```diff
@@ -61,12 +61,69 @@ in
...
+
+        "/holofuel/" = {
+          root = pkgs.holofuel-ui;
+          index = "index.html index.php";
```
Comment (yegortimoshenko) [inline profiles/holoportos/default.nix]:
```suggestion
```

File: profiles/holoportos/default.nix (hunk 2 above)
Comment (zo-el) [reply profiles/holoportos/default.nix]: :+1:

**Pull request merged**